packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -348,15 +349,29 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 		}
 	}
 
-	protected override reSubmitCore(content: any, localOpMetadata: unknown) {
+	protected override reSubmitCore(content: JsonCompatibleReadOnly, localOpMetadata: unknown) {
+		assert(isJsonObject(content), "expected content to be json object");
+		assert(
+			typeof content.revision === "string" && isStableId(content.revision),
+			"expected revision id to be valid stable id",
+		);
+		assert(isStableId(content.revision), "invalid stable ID");
 		const [commit] = this.editManager.findLocalCommit(content.revision);
 		this.submitCommit(commit);
 	}
 
-	protected applyStashedOp(content: any): undefined {
-		const { revision, changeset } = content as Message;
-		const decodedChangeset = this.changeFamily.encoder.decodeJson(formatVersion, changeset);
-		this.addLocalChange(decodedChangeset, revision);
+	protected applyStashedOp(content: JsonCompatibleReadOnly): undefined {
+		assert(isJsonObject(content), "expected content to be json object");
```

Comment:
I'm not clean on the difference between reSubmitCore and applyStashedOp, but it seem like both are parsing persisted data and have duplicated logic, which is also likely duplicated elsewhere for non stashed op.

Generally we want to factor out our logic which deals with parsing of persisted data (in this case ops) into as self contained code as possible since it (and it along) should have to deal with legacy data formats from old content.

Can we combine these two cases (and possible more), and pull them both out into a single function which top level pure function which handles the validation and parsing of this stashed op data format? And include more explicit documentation and robust validation for this data format?

Reply:
I've added a function called `parseStashedOpContent` which takes in a `JsonCompatibleReadOnly`, and returns a typechecked `revision` and `changeset`.

Would this be sufficient? Or were there more specific checks that should also be included?

Reply:
I also noticed that since I imported the `StableId` type and `isStableId` function from the idcompressor, I am now failing the test in `fence.spec.ts`.

Is there a proper way to import them, or is there a simpler way to make these checks without the imports?

Reply:
It might make sense for this function to return the `Commit` type since `Message` is effectively an encoded `Commit`. This would mean this function would also be responsible for decoding the changeset.

This logic can also be used in `processCore`.

Reply:
updated parse function to `parseCommit` which takes in the `content` and `decoder` function, and returns a commit object.